packages/server/lib/middleware/ratelimit.middleware.ts

```diff
@@ -43,25 +44,7 @@ async function getRateLimiter(size: DBPlan['api_rate_limit_size']) {
     const url = getRedisUrl();
     let limiter: RateLimiterAbstract;
     if (url) {
-        const isExternal = url.startsWith('rediss://');
-        const socket = isExternal
-            ? {
-                  reconnectStrategy: (retries: number) => Math.min(retries * 200, 2000),
-                  connectTimeout: 10_000,
-                  tls: true,
-                  servername: new URL(url).hostname,
-                  keepAlive: 60_000
-              }
-            : {};
-        const redisClient = await createClient({
-            url: url,
-            disableOfflineQueue: true,
-            pingInterval: 30_000,
-            socket
-        }).connect();
-        redisClient.on('error', (err) => {
-            logger.error(`Redis (rate-limiter) error: ${err}`);
-        });
+        const redisClient = ((await getKVStore('default', { url })) as KVStoreRedis).getClient();
```

Comment:
why not calling getRedis directly instead of instanciating a KVStore just to use its client?

Reply:
Thought I answered this. I didn't want to expose the getRedis directly so there was only a single entry point to KV functionality.